sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -307,6 +303,151 @@ export class ApplicationService extends RecordDataModelService<Application> {
     return { application, createdAssessment: originalAssessment };
   }
 
+  /**
+   * Starts a change request for an existing student application.
+   * A new application will be created in edited status and will be assessed by the Ministry.
+   * @param applicationId application ID to be changed.
+   * @param studentId student ID.
+   * @param applicationData application dynamic data to be saved.
+   * @param associatedFiles files to be associated with the application.
+   * @param auditUserId user that should be considered the one that is making the change request.
+   * @returns application ID of the created application that represents the change request.
+   */
+  async submitApplicationChangeRequest(
+    applicationId: number,
+    studentId: number,
+    applicationData: ApplicationData,
+    associatedFiles: string[],
+    auditUserId: number,
+  ): Promise<ApplicationSubmissionResult> {
+    const application = await this.getApplicationToSave(
+      studentId,
+      ApplicationStatus.Completed,
+      applicationId,
+    );
+    // If the application was not found it is because it does not belong to the student or it is not completed.
+    if (!application) {
+      throw new CustomNamedError(
+        "Student application not found or it is not in the correct status to be changed.",
+        APPLICATION_NOT_FOUND,
+      );
+    }
+    if (!allowApplicationChangeRequest(application.programYear)) {
+      throw new CustomNamedError(
+        "The program year associated to this application does not allow a change request submission.",
+        APPLICATION_NOT_VALID,
+      );
+    }
+    // Check if there is already a change request in progress.
+    const hasChangeInProgress = application.parentApplication.versions.some(
+      (version) =>
+        [
+          ApplicationEditStatus.ChangeInProgress,
+          ApplicationEditStatus.ChangePendingApproval,
+        ].includes(version.applicationEditStatus),
+    );
```

Comment:
As the versions are joined based on same criteria, we can either check only for versions or remove the criteria in the query.

Reply:
I would like to keep both independently checking it, which would allow the query (that is already shared) to be expanded in the future without the need to impact this method.
If the query was dedicated to this method only, I would be open to it. Since it is a shared query, it would introduce an unnecessary liability.

Reply:
I still feel that we should remove the condition on join from the query for the same reason that it is shared.